pkg/monitoring/metrics/virt-controller/vmistats_collector.go

```diff
@@ -172,68 +172,88 @@ func getGuestOSInfo(vmi *k6tv1.VirtualMachineInstance) (kernelRelease, machine,
 }
 
 func getVMIInstancetype(vmi *k6tv1.VirtualMachineInstance) string {
-	instanceType := none
-
 	if instancetypeName, ok := vmi.Annotations[k6tv1.InstancetypeAnnotation]; ok {
-		instanceType = other
+		return fetchInstancetypeName(instancetypeName)
+	}
 
-		obj, ok, err := instanceTypeInformer.GetIndexer().GetByKey(instancetypeName)
-		if err != nil || !ok {
-			return instanceType
-		}
+	if instancetypeName, ok := vmi.Annotations[k6tv1.ClusterInstancetypeAnnotation]; ok {
+		return fetchClusterInstancetypeName(instancetypeName)
+	}
 
-		vendorName := obj.(*instancetypev1beta1.VirtualMachineInstancetype).Labels[instancetypeVendorLabel]
-		if _, isWhitelisted := whitelistedInstanceTypeVendors[vendorName]; isWhitelisted {
-			instanceType = instancetypeName
-		}
+	return none
+}
+
+func fetchInstancetypeName(instancetypeName string) string {
+	instanceType := other
+
+	obj, ok, err := instanceTypeInformer.GetIndexer().GetByKey(instancetypeName)
+	if err != nil || !ok {
+		return instanceType
 	}
 
-	if instancetypeName, ok := vmi.Annotations[k6tv1.ClusterInstancetypeAnnotation]; ok {
-		instanceType = other
+	vendorName := obj.(*instancetypev1beta1.VirtualMachineInstancetype).Labels[instancetypeVendorLabel]
+	if _, isWhitelisted := whitelistedInstanceTypeVendors[vendorName]; isWhitelisted {
```

Comment:
since metrics are collected to telemetry systems by users and vendors, we try to reduce the amount of values a label can have to reduce [cardinality](https://grafana.com/blog/2022/02/15/what-are-cardinality-spikes-and-why-do-they-matter/)